libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -906,6 +906,8 @@ unsigned CompilerUtils::loadFromMemoryHelper(Type const& _type, bool _fromCallda
 		if (leftAligned)
 			m_context << shiftFactor << Instruction::MUL;
 	}
+	if (_fromCalldata && _type.category() == Type::Category::Bool)
```

Comment:
Because nobody asked for it until now.  I'll try that and see how the tests behave.